client.go

```diff
@@ -17,26 +18,72 @@ type SSHClient interface {
 	For(e *Endpoint) tea.ExecCommand
 }
 
+func proxyJump(addr, nextAddr string, conf, nextConf *gossh.ClientConfig) (*gossh.Client, closers, error) {
+	var cl closers
+	log.Info("connecting client to ProxyJump", "addr", addr)
+	jumpClient, err := gossh.Dial("tcp", addr, conf)
+	if err != nil {
+		return nil, cl, fmt.Errorf("connection to ProxyJump (%q) failed: %w", addr, err)
+	}
+	cl = append(cl, jumpClient.Close)
+
+	log.Info("connecting to target using jump client", "addr", nextAddr)
+	jumpConn, err := jumpClient.Dial("tcp", nextAddr)
+	if err != nil {
+		return nil, cl, fmt.Errorf("connection from ProxyJump (%q) to Host (%q) failed: %w", addr, nextAddr, err)
+	}
+	cl = append(cl, jumpConn.Close)
+
+	log.Info("getting client connection", "addr", nextAddr)
+	ncc, chans, reqs, err := gossh.NewClientConn(jumpConn, nextAddr, nextConf)
+	if err != nil {
+		return nil, cl, fmt.Errorf("client connection from ProxyJump (%q) to Host (%q) failed: %w", addr, nextAddr, err)
+	}
+	cl = append(cl, ncc.Close)
+
+	return gossh.NewClient(ncc, chans, reqs), cl, nil
+}
+
 func createSession(conf *gossh.ClientConfig, e *Endpoint, abort <-chan os.Signal, env ...string) (*gossh.Session, *gossh.Client, closers, error) {
 	var cl closers
 	var conn *gossh.Client
 	var err error
 	connected := make(chan bool, 1)
 
-	go func() {
-		conn, err = gossh.Dial("tcp", e.Address, conf)
-		connected <- true
-	}()
+	if jump := e.ProxyJump; jump == "" {
+		go func() {
+			conn, err = gossh.Dial("tcp", e.Address, conf)
+			connected <- true
+		}()
+	} else {
+		jumpConf := &gossh.ClientConfig{
+			User:            conf.User,
+			Auth:            conf.Auth,
+			HostKeyCallback: conf.HostKeyCallback,
+		}
+		username, addr, ok := strings.Cut(jump, "@")
```

Comment:
This will fail if the username contains an `@` symbol as SSH allows it in the username component